src/backend/aspen/api/schemas/usergroup.py

```diff
@@ -5,6 +5,15 @@
 from aspen.api.schemas.locations import LocationResponse
 
 
+class GroupCreationRequest(BaseRequest):
+    name: str
+    prefix: str
+    address: Optional[str]
```

Comment:
can/should we use constr() on these input strings to restrict length or anything?

Reply:
I think only `prefix` needs to be restricted, but I'm not sure what a good max character length would be.

Reply:
I'll add some reasonable restrictions.

Reply:
I'm a bit late to the conversation, but I wonder if we should add some sort of suffix requirement to `prefix` (heh). Most of our group prefixes end in `-` (eg, `CA-SLOC-`), but it's not a hard and fast rule right now, probably about 35% of our real user groups don't follow the convention. Seems like we should probably enforce a convention here, one way or the other?

... huh... Looks like the code changed at some point. We used to handle this via `create_public_ids` func which handled a dynamic dash, but it appears that function doesn't get called anywhere and the prefix is now directly used in the `generate_public_identifier` now. So if you look at the External Demo Group's account on Prod for an old sample (looking at 2022-03-17 right now), you see `hCoV-19/USA/demo-29624/2022`, but if you look at something more recent (2022-05-03) you get `hCoV-19/USA/demo--34277/2022` because the External Demo Group's group's prefix is `demo-` (not to be confused with DEMO ACCOUNT, which has a prefix of plain `demo`).

While we're discussing it, I wonder if we should also restrict spaces in `prefix`. I think there might be some stuff downstream that assumes public IDs never have a space included. I doubt we'd be crazy enough to enter a space when making a new group's `prefix`, but could be nice to enforce in code since this will be our point of intake from here on out.